File: java-src/alchemist-unity/src/main/java/nodes/GradientNode.java
package nodes;

/***
 * Class that represent a Gradient node.
 * It contains:
 * id: int
 * position: NodePosition2D
 * molecules: GradientMoleculesMap
 */
public class GradientNode {

    private static final NodePosition2D DEF_POS = new NodePosition2D(0, 0);

    private final int id;
    private final NodePosition2D position;
    private final GradientMoleculesMap molecules;


    /***
     * @param id the id you want for this node
     * @param position the position you want for this node
     */
    public GradientNode(final int id, final NodePosition2D position) {
        this.id = id;
        this.position = position;
        molecules = new GradientMoleculesMap();
    }

    /***
     * @param id the id you want for this node
     */
    public GradientNode(final int id) {
        this(id, DEF_POS);
    }

    /***
     * Let you change the position of this node with the one requested.
     * @param pos the new position for this node
     */
    public void setPosition(final NodePosition2D pos) {
        position.setPosx(pos.getPosx());
        position.setPosz(pos.getPosz());
    }

    /***
     * Return the node id.
     * @return the node id
     */
    public int getID() {
        return id;
    }

    /***
     * Return the node position.
     * @return the node position
     */
    public NodePosition2D getPosition() {
        return position;
    }

    /***
     * Return all the molecules of this node.
     * @return the molecules of this node
     */
    public IMoleculesMap getMolecules() {
        return molecules;
    }

    /***
     * Let you set a molecule concentration.
     * @param mol the molecule name
     * @param conc the concentration of that molecule
     */
    public void setMolecule(final String mol, final Object conc) {
        molecules.setMolecule(mol, conc);
    }
}

File: unity3d-src/Examples/SmartCity/Assets/Scripts/Node/SimNodeImplementations/GradientNode.cs
using System;
using UnityEngine;

/// <summary>
/// 
/// Class representing an implementation of a Unity version of an Alchemist node, in particular of a node performing a gradient
/// program.
/// 
/// </summary>
[Serializable]
public class GradientNode : ISimNode{

    [SerializeField]
	private int id;
    [SerializeField]
    private NodePosition2D position;
    [SerializeField]
    private GradientMoleculesMap molecules;

    private AbstractBehaviourNode panel;

    public GradientNode(int id, NodePosition2D position, AbstractBehaviourNode panel)
	{
		this.id = id;
        this.position = position;
        this.panel = panel;
        this.molecules = new GradientMoleculesMap();
    }
    
    public int GetID()
    {
        return id;
    }

    public NodePosition2D GetPosition()
    {
        return position;
    }

    public void SetPosition(NodePosition2D pos)
    {
        position.SetPosx(pos.GetPosx());
        position.SetPosz(pos.GetPosz());
    }

    public GradientMoleculesMap GetMolecules()
    {
        return molecules;
    }

    public void SetMolecule(string mol, object conc)
    {
        molecules.SetMolecule(mol, conc);
    }

    public void SetMolecules(GradientMoleculesMap molecules)
    {
        this.molecules.SetMolecule("data", (double)molecules.GetMoleculeConcentration("data"));
    }

    public AbstractBehaviourNode GetGameObject()
    {
        return panel;
    }
}
